apps/prairielearn/assets/scripts/instructorQuestionSettingsClient.ts

```diff
@@ -11,6 +19,46 @@ onDocumentReady(() => {
   );
   const saveButton = document.querySelector<HTMLButtonElement>('#save-button');
 
+  new TomSelect('#topic', {
+    valueField: 'name',
+    searchField: ['name'],
+    closeAfterSelect: true,
+    items: [(document.querySelector('#topic') as HTMLSelectElement).dataset.selectedTopic],
+    plugins: ['dropdown_input'],
+    maxItems: 1,
+    placeholder: 'Select a topic',
+    render: {
+      option(data: TopicData, escape: (input: string) => string) {
+        return (
+          '<div>' +
+          '<span class="badge justify-content-start color-' +
+          escape(data.color) +
+          '">' +
+          escape(data.name) +
+          '</span>' +
```

Comment:
Can you reuse the `TopicBadge` component here to avoid duplicating markup?

On that note, because this function is just expected to return a string, this whole thing can be constructed with our usual `html` tagged-template literal. Just make sure to `.toString()` it at the end. Ditto for `item(...)` below.

Reply:
Yes, updated to use the `TopicBadge` and our `html` template literal in [98f2dd5](https://github.com/PrairieLearn/PrairieLearn/pull/10868/commits/98f2dd558c3f286228530e2a7899b8db990394be).